README.md

```diff
@@ -1,2 +1,484 @@
-# browserstack-examples-testng
-This repository showcases the BrowserStack integration features with the Java TestNG using the BrowserStack sample demo canonical app. More details: https://browserstack.atlassian.net/wiki/spaces/CE/pages/2091057195/Customer+Demo+Repositories
+![Logo](https://www.browserstack.com/images/static/header-logo.jpg)
+
+# BrowserStack Examples TestNG
+
+## Introduction
+
+TestNG is a testing framework inspired from JUnit and NUnit but introducing some new functionalities that make it more powerful.
+
+This BrowserStack Example repository demonstrates a Selenium test framework written in TestNG with parallel testing capabilities. The Selenium test scripts are written for the open source [BrowserStack Demo web application](https://bstackdemo.com) ([Github](https://github.com/browserstack/browserstack-demo-app)). This BrowserStack Demo App is an e-commerce web application which showcases multiple real-world user scenarios. The app is bundled with offers data, orders data and products data that contains everything you need to start using the app and run tests out-of-the-box.
+
+The Selenium tests are run on different platforms like on-prem, docker and BrowserStack using various run configurations and test capabilities.
+
+---
+
+## Repository setup
+
+- Clone the repository
+
+- Ensure you have the following dependencies installed on the machine
+   - Java >= 8
+   - Maven >= 3.1+
+
+  Maven:
+    ```sh
+     mvn install -DskipTests
+    ```
+
+  Gradle:
+
+    ```sh
+     gradle clean build
+    ```
+
+
+## About the tests in this repository
+
+This repository contains the following Selenium tests:
+
+| Module   | Test name                          | Description |
+  | ---   | ---                                   | --- |
+| E2E      | OrderTest.class               | This test scenario verifies successful product purchase lifecycle end-to-end. It demonstrates the [Page Object Model design pattern](https://www.browserstack.com/guide/page-object-model-in-selenium) and is also the default test executed in all the single test run profiles. |
+| Login    | LoginTest.class               | This test verifies the login workflow with different types of valid login users. |
+| Login    | LoginFailTest.class           | This test verifies the login workflow error for a locked user. |
+| Offers   | OfferTest.class               | This test mocks the GPS location for Mumbai and verifies that the product offers applicable for the Mumbai location are shown.   |
+| Product  | FilterTest.class              | This test verifies that the Apple products are only shown if the Apple vendor filter option is applied. |
+| Product  | FilterTest.class              | This test verifies that the product prices are in ascending order when the product sort "Lowest to Highest" is applied. |
```

Comment:
lets not repeat entries. if multiple tests have been written in the same file, update the description accordingly